### PR TITLE
Core Profiler - replace install-async with install-activate-async

### LINKED ISSUE #38410: Core Profiler - update install-async endpoint to activate plugins
This is related to [API: Create an API that returns status and errors of a queued plugin installation job](https://github.com/woocommerce/team-ghidorah/issues/212#top) issue.

In the original issue, we've added [install-async](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/OnboardingPlugins.php#L51) endpoint to install plugins in the action scheduler.

It turned out that we also needed to activate plugins. 

Update the endpoint to install and activate requested plugins.

### PR DESCRIPTION
**Pull request by moon0326**
### Changes proposed in this Pull Request:

Closes #38410 

This PR replaces `install-async` endpoint with `install-and-activate-async` to activate installed plugins.

### How to test the changes in this Pull Request:

1. Make a request to POST `/wp-json/wc-admin/onboarding/plugins/install-and-activate-async` with the following JSON request

```
{
    "plugins": ["test"]
}
```

2. The response should contain `job_id` field. Copy it.
3. Make a request to GET `/wp-json/wc-admin/onboarding/plugins/scheduled-installs/:job_id`. It should return the following response.

```
{
    "job_id": ":job_id",
    "status": "complete",
    "plugins": {
        "test": {
            "status": "failed",
            "errors": [
                "The requested plugin `test` could not be installed. Plugin API call failed."
            ],
            "install_duration": 0
        }
    }
}
```

4. Repeat steps 1 ~ 3, but with a valid plug-in name this time. When everything goes well, `plugins.:name.status` field value should be `activated`

### CODE REVIEW

File: plugins/woocommerce/src/Admin/PluginsHelper.php [hunk 1]
```diff
@@ -330,6 +330,7 @@ public function install_plugins_async_callback( array $plugins, string $job_id )
 		$option_name = 'woocommerce_onboarding_plugins_install_async_' . $job_id;
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/PluginsHelper.php:330]: Should we also rename this function to `install_and_activate_plugins_async_callback`?

And I'm seeing this error in error.log. Should we make it static?

```
[25-May-2023 08:06:50 UTC] PHP Fatal error:  Uncaught TypeError: call_user_func_array(): Argument #1 ($callback) must be a valid callback, non-static method Automattic\WooCommerce\Admin\PluginsHelper::install_plugins_async_callback() cannot be called statically in /var/www/html/wp-includes/class-wp-hook.php:308
Stack trace:
#0 /var/www/html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#1 /var/www/html/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#2 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/actions/ActionScheduler_Action.php(43): do_action_ref_array('woocommerce_plu...', Array)
#3 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php(65): ActionScheduler_Action->execute()
#4 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(166): ActionScheduler_Abstract_QueueRunner->process_action(102, 'WP Cron')
#5 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(136): ActionScheduler_QueueRunner->do_batch(25, 'WP Cron')
#6 /var/www/html/wp-includes/class-wp-hook.php(308): ActionScheduler_QueueRunner->run('WP Cron')
#7 /var/www/html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#8 /var/www/html/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#9 /var/www/html/wp-cron.php(188): do_action_ref_array('action_schedule...', Array)
#10 /var/www/html/wp-includes/cron.php(886): include_once('/var/www/html/w...')
#11 /var/www/html/wp-includes/cron.php(1002): spawn_cron(1685002010.5754)
#12 /var/www/html/wp-includes/class-wp-hook.php(308): _wp_cron('')
#13 /var/www/html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array)
#14 /var/www/html/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#15 /var/www/html/wp-settings.php(645): do_action('wp_loaded')
#16 /var/www/html/wp-config.php(258): require_once('/var/www/html/w...')
#17 /var/www/html/wp-load.php(50): require_once('/var/www/html/w...')
#18 /var/www/html/wp-blog-header.php(13): require_once('/var/www/html/w...')
#19 /var/www/html/index.php(17): require('/var/www/html/w...')
#20 {main}
```

Comment (chihsuan) [inline plugins/woocommerce/src/Admin/PluginsHelper.php]: Should we also rename this function to `install_and_activate_plugins_async_callback`?

And I'm seeing this error in error.log. Should we make it static?

```
[25-May-2023 08:06:50 UTC] PHP Fatal error:  Uncaught TypeError: call_user_func_array(): Argument #1 ($callback) must be a valid callback, non-static method Automattic\WooCommerce\Admin\PluginsHelper::install_plugins_async_callback() cannot be called statically in /var/www/html/wp-includes/class-wp-hook.php:308
Stack trace:
#0 /var/www/html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#1 /var/www/html/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#2 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/actions/ActionScheduler_Action.php(43): do_action_ref_array('woocommerce_plu...', Array)
#3 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php(65): ActionScheduler_Action->execute()
#4 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(166): ActionScheduler_Abstract_QueueRunner->process_action(102, 'WP Cron')
#5 /var/www/html/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_QueueRunner.php(136): ActionScheduler_QueueRunner->do_batch(25, 'WP Cron')
#6 /var/www/html/wp-includes/class-wp-hook.php(308): ActionScheduler_QueueRunner->run('WP Cron')
#7 /var/www/html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#8 /var/www/html/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#9 /var/www/html/wp-cron.php(188): do_action_ref_array('action_schedule...', Array)
#10 /var/www/html/wp-includes/cron.php(886): include_once('/var/www/html/w...')
#11 /var/www/html/wp-includes/cron.php(1002): spawn_cron(1685002010.5754)
#12 /var/www/html/wp-includes/class-wp-hook.php(308): _wp_cron('')
#13 /var/www/html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters(NULL, Array)
#14 /var/www/html/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#15 /var/www/html/wp-settings.php(645): do_action('wp_loaded')
#16 /var/www/html/wp-config.php(258): require_once('/var/www/html/w...')
#17 /var/www/html/wp-load.php(50): require_once('/var/www/html/w...')
#18 /var/www/html/wp-blog-header.php(13): require_once('/var/www/html/w...')
#19 /var/www/html/index.php(17): require('/var/www/html/w...')
#20 {main}
```

Comment (moon0326) [reply plugins/woocommerce/src/Admin/PluginsHelper.php]: @chihsuan  Thanks!

I've renamed action and callback in https://github.com/woocommerce/woocommerce/pull/38434/commits/1e09a4d64b5d84fdd4a50eff593c0e5bcb9bb24d

Interesting..I did not get the error in my local. I've changed callback to static in the commit 👍

**Review by chihsuan — APPROVED** [review]
LGTM and tested well. 👍

**Pull request merged**